### ISSUE TITLE
SQLPad with Clickhouse cannot load database schema

### ISSUE THREAD

**Issue by vumdao** [sqlpad/sqlpad#929]
ClickHouse client version 20.3.9.70 (official build).
SQLPad version: sqlpad/sqlpad:6.1.2

![image](https://user-images.githubusercontent.com/37215642/102965501-835b4980-4520-11eb-8986-4e7dcac15f74.png)

When clicking refresh schema the UI shows nothing but clickhouse-server logs shows

`2020.12.23 04:49:42.624767 [ 106 ] {} <Trace> HTTPHandler-factory: HTTP Request for HTTPHandler-factory. Method: POST, Address: 10.2.0.221:50032, User-Agent: node-fetch/1.0 (+https://github.com/bitinn/node-fetch), Length: 291, Content Type: text/plain;charset=UTF-8, Transfer Encoding: identity

2020.12.23 04:49:42.624883 [ 106 ] {} <Trace> HTTPHandler: Request URI: /?database=mydatabase
2020.12.23 04:49:42.625275 [ 106 ] {97ee8c36-6bde-4d1e-8193-ea6c78e7df88} <Debug> executeQuery: (from 10.2.0.221:50032)      SELECT        database as table_schema,        table as table_name,        name as column_name,        type as data_type      FROM        system.columns     WHERE database NOT IN ('system')     AND database = 'mydatabase'     ORDER BY       table_schema,       table_name   
2020.12.23 04:49:42.625857 [ 106 ] {97ee8c36-6bde-4d1e-8193-ea6c78e7df88} <Trace> AccessRightsContext (default): Access granted: SELECT(database, table, name, type) ON system.columns
2020.12.23 04:49:42.630088 [ 106 ] {97ee8c36-6bde-4d1e-8193-ea6c78e7df88} <Trace> InterpreterSelectQuery: FetchColumns -> Complete
2020.12.23 04:49:42.630311 [ 106 ] {97ee8c36-6bde-4d1e-8193-ea6c78e7df88} <Trace> AccessRightsContext (default): Access granted: SHOW ON *.*
2020.12.23 04:49:42.649938 [ 106 ] {97ee8c36-6bde-4d1e-8193-ea6c78e7df88} <Information> executeQuery: Read 10151 rows, 1011.71 KiB in 0.025 sec., 412590 rows/sec., 40.16 MiB/sec.
2020.12.23 04:49:42.650158 [ 106 ] {97ee8c36-6bde-4d1e-8193-ea6c78e7df88} <Debug> MemoryTracker: Peak memory usage (total): 4.06 MiB.
2020.12.23 04:49:42.654289 [ 106 ] {97ee8c36-6bde-4d1e-8193-ea6c78e7df88} <Debug> MemoryTracker: Peak memory usage (for query): 4.06 MiB.
2020.12.23 04:49:42.654405 [ 106 ] {} <Information> HTTPHandler: Done processing query
`

![image](https://user-images.githubusercontent.com/37215642/102965818-33c94d80-4521-11eb-954b-7b420f8d5f07.png)

**Comment by gangtao** [comment]
I am using docker images and have the same issue

**Comment by rickbergfalk** [comment]
@gangtao could you post sqlpad and clickhouse version you are using specifically?

**Comment by vumdao** [comment]
@gangtao It might be problem from sqlpad which cannot load schema of the clickhouse database but we still can run the query to get data, @rickbergfalk can you please help to check?

I use ClickHouse client version 20.3.9.70 (official build) and sqlpad 6.2.1 either 6.1.2

![2021-03-05_18-06](https://user-images.githubusercontent.com/37215642/110107507-8bb3e880-7ddd-11eb-9f78-24b1b2987fa2.png)
![2021-03-05_15-31](https://user-images.githubusercontent.com/37215642/110107525-940c2380-7ddd-11eb-8f11-289ae1fe21f9.png)
![2021-03-05_18-04](https://user-images.githubusercontent.com/37215642/110109742-6f657b00-7de0-11eb-8138-56ab80f5b409.png)
![2021-03-05_18-05](https://user-images.githubusercontent.com/37215642/110109759-74c2c580-7de0-11eb-9c5c-a0009e07366c.png)

**Comment by rickbergfalk** [comment]
I did some digging and know what is going on here. 

In SQLPad 6.1.2 fixes were made to SQLPad's Clickhouse driver implementation that relies on the `X-ClickHouse-Format` header, which was added in Clickhouse version `20.7`.

I'll look into some options. It looks like there are some libraries available now that might support old and new versions of Clickhouse instead of relying on our own implementation
